Source/Immutable/Private/Immutable/ImmutablePassport.cpp

```diff
@@ -428,60 +432,78 @@ void UImmutablePassport::OnInitDeviceFlowResponse(FImtblJSResponse Response)
 
 void UImmutablePassport::OnLogoutResponse(FImtblJSResponse Response)
 {
-	if (auto ResponseDelegate = GetResponseDelegate(Response))
+	auto ResponseDelegate = GetResponseDelegate(Response);
+	
+	if (!ResponseDelegate)
 	{
-		if (Response.success)
-		{
-			FString Url;
-			FString Err;
+		return;
+	}
 
-			Response.JsonObject->TryGetStringField(TEXT("result"), Url);
-			if (!Url.IsEmpty())
-			{
+	FString Message;
+
+	if (!Response.success)
+	{
+		Message = Response.Error.IsSet() ? Response.Error->ToString() : Response.JsonObject->GetStringField(TEXT("error"));
+
+		IMTBL_ERR("%s", *Message)
+		ResponseDelegate->ExecuteIfBound(FImmutablePassportResult{ Response.success, Message, Response });
+
+		return;
+	}
+
+	if (!IsStateFlagsSet(IPS_HARDLOGOUT))
+	{
+		Message = "Logged out skipping Auth0";
```

Comment:
Maybe avoid Auth0 in the message. Use browser session instead. Can you also rewrite the message?